packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -769,19 +769,17 @@ class AppInstance {
 
 /// A [Logger] which sends log messages to a listening daemon client.
```

Comment:
`logToParent` changes the behavior to using the parent to handle all stdout. This is valuable for IntelliJ's implementation of the flutter run -v option - previously we weren't passing trace level messages to the IDEs. An equivalent change would be to keep the current behavior - not add logToParent - but to pass all message levels, including trace, to IDEs.

> why bother creating the object at all?

With or without those changes, this class sends progress messages to the IDE, so it can use that info to provide nice, integrated progress UIs.